src/janitor.c

```diff
@@ -392,6 +394,19 @@ static void pool_client_maint(PgPool *pool)
 		}
 	}
 
+	/* check user client_idle_timout */
+	if (any_user_level_client_timeout_set) {
+		statlist_for_each_safe(item, &pool->active_client_list, tmp) {
```

Comment:
> Most of the more granular settings appear to replace the global level settings. I realized this after raising the PR. If you want this can be reworked to act in the same manner.

Yeah, let's merge this with the loop above so that the user level setting replaces the global one.

Reply:
Agreed. Will do.

Reply:
Merged the loops as described and added additional tests to cover that the user level setting overrides the global level setting.